test/utils.py

```diff
@@ -120,15 +116,66 @@ def get_platform_tags(python_abi_tag):
     return templates
 
 
+def generate_project(path, 
```

Comment:
An alternate version of `project_generator.py` - my thinking is that `project_generator.py` would replace this function